datafusion/expr/src/built_in_function.rs

```diff
@@ -202,6 +203,120 @@ pub enum BuiltinScalarFunction {
     ArrowTypeof,
 }
 
+lazy_static! {
+    /// Mapping between SQL function names to `BuiltinScalarFunction` types.
+    /// Note that multiple SQL function names can represent the same `BuiltinScalarFunction`. These are treated as aliases.
+    /// In case of such aliases, the first SQL function name in the vector is used when displaying the function.
+    static ref NAME_TO_FUNCTION: Vec<(&'static str, BuiltinScalarFunction)> = vec![
```

Comment:
This is a really nice solution and writeup -- thank you @2010YOUY01 

I am slightly concerned about having to do a linear search for each function name, but I don't think it is a big deal and I will propose a follow on PR to improve the situation